p2pserver/link/link.go

```diff
@@ -22,31 +22,37 @@ import (
 	"bufio"
 	"errors"
 	"fmt"
-	"net"
-	"time"
-
+	lru "github.com/hashicorp/golang-lru"
 	comm "github.com/ontio/ontology/common"
 	"github.com/ontio/ontology/common/log"
 	"github.com/ontio/ontology/p2pserver/common"
 	"github.com/ontio/ontology/p2pserver/message/types"
+	"net"
+	"time"
 )
 
 //Link used to establish
 type Link struct {
-	id        uint64
-	addr      string                 // The address of the node
-	conn      net.Conn               // Connect socket with the peer node
-	port      uint16                 // The server port of the node
-	time      time.Time              // The latest time the node activity
-	recvChan  chan *types.MsgPayload //msgpayload channel
-	reqRecord map[string]int64       //Map RequestId to Timestamp, using for rejecting duplicate request in specific time
+	id         uint64
+	addr       string                 // The address of the node
+	conn       net.Conn               // Connect socket with the peer node
+	port       uint16                 // The server port of the node
+	time       time.Time              // The latest time the node activity
+	recvChan   chan *types.MsgPayload //msgpayload channel
+	reqRecord  map[string]int64       //Map RequestId to Timestamp, using for rejecting duplicate request in specific time
+	txMsgCache *lru.Cache             //Map txHash to bool, using for rejecting duplicate tx
```

Comment:
every remote peer has a link, so if N peers connected, it will allocate N Caches. may need move it to transaction handler.